### PR TITLE
aws: remove exceptions from data plane

### PR DESCRIPTION
**Pull request by nbaws**
Commit Message: aws: remove exceptions from data plane
Additional Description: 
Risk Level: Low
Testing: Unit
Docs Changes: N/A
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

@suniltheta @alyssawilk

### CODE REVIEW

**Comment by phlax** [comment]
ci fail looks real

/wait

**Comment by nbaws** [comment]
@phlax @mattklein123 any idea why source/common/json is coverage flagged here? It wasn't changed in this PR.

**Comment by nbaws** [comment]
> @phlax @mattklein123 any idea why source/common/json is coverage flagged here? It wasn't changed in this PR.

I'm guessing it's because I removed calls to getString as they throw. I can't see a test case for this so I'll add one, let me know if you want it included somewhere else than this PR.

File: test/common/json/json_loader_test.cc [hunk 1]
```diff
@@ -78,7 +78,8 @@ TEST_F(JsonLoaderTest, Basic) {
     EXPECT_THROW(json->getBoolean("hello"), Exception);
     EXPECT_THROW(json->getObjectArray("hello"), Exception);
     EXPECT_THROW(json->getString("hello"), Exception);
-
+    EXPECT_THROW_WITH_MESSAGE(json->getString("hello",""), Exception,
```
Comment (nbaws) [inline test/common/json/json_loader_test.cc:81]: Added test case for getString with default. Appears to be no users outside of contrib any more.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**